fdbserver/EncryptKeyProxy.actor.cpp

```diff
@@ -150,19 +154,37 @@ struct EncryptKeyProxyData : NonCopyable, ReferenceCounted<EncryptKeyProxyData>
 	}
 };
 
+struct pair_hash {
+	template <class T1, class T2>
+	std::size_t operator()(const std::pair<T1, T2>& pair) const {
+		auto hash1 = std::hash<T1>{}(pair.first);
+		auto hash2 = std::hash<T2>{}(pair.second);
+
+		// Equal hashes XOR would be ZERO.
+		return hash1 == hash2 ? hash1 : hash1 ^ hash2;
```

Comment:
Maybe use `boost::hash_combine` instead?

Reply:
You can also use boost::hash(std::pair<...>) directly in your unordered_map specification. See for example:

https://github.com/apple/foundationdb/blob/b8fbd18199c0893a72f8a46fead3a22c1940bd3f/fdbclient/DatabaseContext.h#L634